### ISSUE TITLE
operations that select fields from `extend type Query` fail to validate

### ISSUE THREAD

**Issue by lennyburdette** [apollographql/apollo-rs#440]
## Description

I'm getting incorrect diagnostics when trying to select fields defined on a type extension.

### Steps to reproduce

```rs
let mut compiler = ApolloCompiler::new();
compiler.add_type_system(
    r#"
type Query {
  foo: String
}

extend type Query {
  bar: String
}
"#,
    "schema.graphql",
);

compiler.add_executable("query Query { foo bar }", "op.graphql");
let diagnostics = compiler.validate();
for diagnostic in &diagnostics {
    println!("{}", diagnostic);
}
```

### Expected result

No diagnostics.

### Actual result

```
  × Cannot query `bar` field
   ╭────
 1 │ query Query { foo bar }
   ·                   ──┬─
   ·                     ╰── `bar` field is not in scope
   ╰────
  help: `bar` is not defined on `Query` type
```

### Environment

- Crate version: 0.6.0

**Comment by lrlna** [comment]
yep, this one is legit - we are still add extensions to our validation rules! coming super soon though

**Comment by lrlna** [comment]
this one is fixed with some of the work in #482 and #484, and is published as part of the new releases. `cargo add apollo-compiler@0.7.1`
